packages/proxy/lib/http/util/prerequests.ts

```diff
@@ -330,4 +361,27 @@ export class PreRequests {
     this.pendingRequests = new QueueMap<PendingRequest>()
     this.pendingUrlsWithoutPreRequests = new QueueMap<PendingUrlWithoutPreRequest>()
   }
+
+  private static hasServiceWorkerHeader (headers: Record<string, string | string[] | undefined>) {
+    return headers?.['service-worker'] === 'script' || headers?.['Service-Worker'] === 'script'
+  }
+
+  private shouldIgnorePendingRequest (browserPreRequest: BrowserPreRequest) {
```

Comment:
It feels like this check & the serviceworker pass-throughs should be de-coupled from the prerequests this file and  moved into `http/index.ts` to pass directly to the service worker manger.

https://github.com/cypress-io/cypress/blob/ryanm/fix/service-worker-capture/packages/proxy/lib/http/index.ts#L442-L443
``` diff
  // http/index.ts
  addPendingBrowserPreRequest (browserPreRequest: BrowserPreRequest) {
+    if (!this.serviceWorkerManager.willHandleRequest(browserPreRequest)) {
      this.preRequests.addPending(browserPreRequest)
+    }
  }
...
  reset () {
    this.buffers.reset()
    this.setAUTUrl(undefined)
    this.preRequests.reset()
+  this.serviceWorkerManager = new ServiceWorkerManager()
  }
```

Reply:
I like this idea. Resolved here: [`04617eb` (#28517)](https://github.com/cypress-io/cypress/pull/28517/commits/04617eb1b3565c380ac89d12f575f2e731cba4f6)